tflint/runner.go

```diff
@@ -186,7 +188,7 @@ func NewModuleRunners(parent *Runner) ([]*Runner, error) {
 			}
 		}
 
-		runner, err := NewRunner(parent.config, parent.annotations, cfg)
+		runner, err := NewRunner(parent.config, parent.files, parent.annotations, cfg)
```

Comment:
In the module inspection, the second argument should be the files of the module to be parsed, not the parent module. This means that we need to use Loader here.

However, it would not be a good design for Runner to rely on Loader internally. One idea is to preload all files for all modules on the first load. Maybe it's a bit of a daunting task, what do you think?

Reply:
Ooh yes good catch. I think this map will actually contain all files loaded by the loader, since `BuildConfig` will walk all of the child modules:

https://github.com/terraform-linters/tflint/blob/be3aa7bbfe941ca64d7b0cfdb50d676605630171/tflint/loader.go#L97

After doing a bit of poking, I don't think this will break anything, since any path references Terraform has should be relative to the parent dir. However, it does mean that a runner can potentially return a file used outside of its module.

Rather than try to call [`parser.ConfigDirFiles`](https://pkg.go.dev/github.com/hashicorp/terraform/configs?tab=doc#Parser.ConfigDirFiles) and essentially mix in loader behavior to the runner, a simple fix here would be to continue to use the same map of files in every runner but validate that path provided has the same directory as the module.

Reply:
Oops, I was misunderstanding. I thought this map would only contain root module files. As you say, it seems that all files are included.

Certainly, it's possible that we can refer to files outside the module, but I don't think there's a big problem. Thank you for the explanation.